charts/api/templates/deployment-queue.yaml

```diff
@@ -1,32 +1,34 @@
+{{- range .Values.queue.queueNames }}
+---
 apiVersion: apps/v1
 kind: Deployment
 metadata:
-  name: {{ include "api.fullname" . }}-queue
+  name: {{ include "api.fullname" $ }}-queue-{{ . }}
   labels:
-{{ include "api.labels" . | indent 4 }}
+{{ include "api.labels" $ | indent 4 }}
```

Comment:
`range` shifts the context (`.`) within the loop to the queue name, making the outer context available as `$`, hence all the noise.